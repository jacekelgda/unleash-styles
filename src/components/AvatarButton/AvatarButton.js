/** @flow */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './AvatarButton.css'

/**
 * Avatar component which can be used in relation to user profile
 */
class AvatarButton extends Component {
  static defaultProps: Object

  handleClick () {
    const { onClick } = this.props
    onClick()
  }

  render () {
    return (
      <div>
        <input
          type='checkbox'
          onClick={this.handleClick}
          className={styles.AvatarButton}
          id='avatar_button'
        />
        <label
          className={styles.AvatarButtonLabel}
          htmlFor='avatar_button'>{this.props.label}
        </label>
      </div>
    )
  }
}

AvatarButton.defaultProps = {
  label: null,
  onClick: val => val
}

AvatarButton.propTypes = {
  /** Label will be used for displaying text on avatar */
  label: PropTypes.string,
  /** Function to be called after clicking avatar */
  onClick: PropTypes.func
}

export default AvatarButton
