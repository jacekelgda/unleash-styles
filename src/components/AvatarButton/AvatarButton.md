AvatarButton component:

    <AvatarButton label="ON" />
