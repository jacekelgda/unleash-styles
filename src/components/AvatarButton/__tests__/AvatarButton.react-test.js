/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'

import AvatarButton from '../AvatarButton'

test('AvatarButton renders correctly', () => {
  const wrapper = shallow(<AvatarButton label='test' />)
  expect(wrapper).toMatchSnapshot()
})
