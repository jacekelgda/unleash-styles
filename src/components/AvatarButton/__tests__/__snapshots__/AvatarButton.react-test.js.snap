// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarButton renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <input
        className="AvatarButton"
        id="avatar_button"
        onClick={[Function]}
        type="checkbox"
    />
    <label
        className="AvatarButtonLabel"
        htmlFor="avatar_button"
    >
        test
    </label>
</div>,
  "nodes": Array [
    <div>
      <input
            className="AvatarButton"
            id="avatar_button"
            onClick={[Function]}
            type="checkbox"
      />
      <label
            className="AvatarButtonLabel"
            htmlFor="avatar_button"
      >
            test
      </label>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AvatarButton
        label="test"
        onClick={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AvatarButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "label": "test",
          "onClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <input
                    className="AvatarButton"
                    id="avatar_button"
                    onClick={[Function]}
                    type="checkbox"
          />
          <label
                    className="AvatarButtonLabel"
                    htmlFor="avatar_button"
          >
                    test
          </label>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <input
                    className="AvatarButton"
                    id="avatar_button"
                    onClick={[Function]}
                    type="checkbox"
          />
          <label
                    className="AvatarButtonLabel"
                    htmlFor="avatar_button"
          >
                    test
          </label>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AvatarButton
    label="test"
    onClick={[Function]}
/>,
}
`;
